### ISSUE TITLE
Redundant JOIN in simple Where for one-to-many association

### ISSUE THREAD

**Issue by nettashamir-allocate** [dotnet/efcore#32419]
## Description
For a simple one-to-many association, eg **Account** has many **Blogs**, I would expect a linq query of the form:

```
context.Blogs.Where( blog => blog.Account == myAccount)
```
or
```
context.Blogs.Where( blog => blog.Account.ID == myAccountID)
```

to generate SQL of the form:

```
SELECT <columns> 
FROM [Blog] AS [b] 
WHERE [b].AccountID = <myAccountID>
```

but instead I get:

```
SELECT <columns>
FROM [Blog] AS [b]
INNER JOIN [Account] AS [a] ON [b].[AccountID] = [a].[AccountID]
WHERE [a].[AccountID] = <myAccountID>
```

The INNER JOIN seems to be redundant, and, more importantly, is twice as expensive.

Looking around, I've seen other posts which suggest adding a mapped ID property for AccountID which does indeed produce the expect SQL.
However this seems redundant (and ugly). It would also mean having to make changes throughout a very large body of code for all places which currently do this kid of query by one of the other two methods.

Is there something I'm missing or is this the only way to achieve the efficient SQL?

### Full Repro Code

```C#
public class Account {
	protected int id;
	public int ID { get { return id; } }
	public IEnumerable<Blog> Blogs { get; set; }
}

public class AccountMappings : IEntityTypeConfiguration<Account> {
	public void Configure(EntityTypeBuilder<Account> entity) {
		entity.ToTable("Account");
		entity.HasKey(e => e.ID);
		entity.Property(e => e.ID).HasColumnName("AccountID").HasField("id");
	}
}

public class Blog {
	protected int id;
	public int ID { get { return id; } }
	public Account Account { get; set; }

	// Explicitly declared ID property - Don't want to have to do this...
	public int AccountID { get; set; }
}

public class BlogMappings : IEntityTypeConfiguration<Blog> {
	public void Configure(EntityTypeBuilder<Blog> entity) {
		entity.ToTable("Blog");
		entity.HasKey(e => e.ID);
		entity.Property(e => e.ID).HasColumnName("BlogID").HasField("id");
		entity.HasOne(e => e.Account).WithMany(a => a.Blogs);
		entity.Property(e => e.AccountID);
	}
}

public class TestDbContext : DbContext {
	public DbSet<Account> Accounts { get; set; }
	public DbSet<Blog> Blogs { get; set; }

	protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder) {
		optionsBuilder	.LogTo(Console.WriteLine, LogLevel.Information).UseSqlServer(CONNECTION_STRING);
	}
	protected override void OnModelCreating(ModelBuilder modelBuilder) {
		modelBuilder.ApplyConfiguration(new AccountMappings());
		modelBuilder.ApplyConfiguration(new BlogMappings());
	}
}
```
Tests
```C#
public class ReproTests {
	private TestDbContext context;

	[OneTimeSetUp]
	public void OneTimeSetUp() {
		context = new TestDbContext();
		context.Database.EnsureCreated();
	}

	[Test]
	public void GetByID_SqlIncludesRedundantJoin() {
		_ = context.Blogs.Where(b => b.Account.ID == 1).ToList();
	}

	[Test]
	public void GetByEntityEqualityCheck_SqlIncludesRedundantJoin() {
		var dummyAccount = new Account();
		_ = context.Blogs.Where(b => b.Account == dummyAccount).ToList();
	}

	[Test]
	public void GetByMappedID_SqlDoesNotIncludeRedundantJoin() {
		_ = context.Blogs.Where(b => b.AccountID == 1).ToList();
	}
}

```
### Generated SQL

For **GetByID_SqlIncludesRedundantJoin** and **GetByEntityEqualityCheck_SqlIncludesRedundantJoin**:
``` 
      SELECT [b].[BlogID], [b].[AccountID]
      FROM [Blog] AS [b]
      INNER JOIN [Account] AS [a] ON [b].[AccountID] = [a].[AccountID]
      WHERE [a].[AccountID] = 1
```
For **GetByMappedID_SqlDoesNotIncludeRedundantJoin**:
```
      SELECT [b].[BlogID], [b].[AccountID]
      FROM [Blog] AS [b]
      WHERE [b].[AccountID] = 1
```

### Include provider and version information

EF Core version: 7.0.5
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 6.0
Operating system: WIndows 11
IDE: Visual Studio 2022 17.7

**Comment by roji** [comment]
Duplicate of #28077

**Comment by roji** [comment]
Thanks @nettashamir-allocate, I'm pretty sure this is a duplicate of #28077. Yes, I agree that if only the key of the related entity is accessed, then simple accessing the foreign key is much preferable to adding a whole join.

**Comment by roji** [comment]
Re #28077, I'll put a consider-for-current-release as a query perf issue, though I can't say whether we'll have enough time to fix it for 9.0...
